provisioning/transport/http/test/_http_test.js

```diff
@@ -45,7 +46,25 @@ describe('Http', function () {
     http.setTransportOptions({
       pollingInterval: 1
     });
-    return http;
+
+    var pollingStateMachine = new PollingStateMachine(http);
+
+    pollingStateMachine.registerX509 = function(request, auth, callback) {
+      http.setAuthentication(auth);
+      pollingStateMachine.register(request, function(err, body, result) {
+        if (err) {
+          callback(err);
+        } else {
+          callback(null, body.registrationState, body);
+        }
+      });
+    };
+
+    pollingStateMachine.setTransportOptions = function(options) {
+      http.setTransportOptions(options);
+    }
+
+    return pollingStateMachine;
```

Comment:
I have a couple of worries for this thing:
1. it creates something called pollingStateMachine and returns it when the function is called makeNewTransport and those are separate concepts (maybe what is meant here is PollingTransport?)

2. it "hides" 2 API calls behind one that looks like an old one.

I'm afraid a couple months from now when we return to this for a bug or something it's going to be very confusing. To be honest it's a little confusing now already. I would much rather see the complete API flow in the tests, even if it's a little repetitive.

Reply:
I agree.  I'm going to create a TFS task for this and a few other items from this review

---
In reply to: [155611534](https://github.com/Azure/azure-iot-sdk-node/pull/176#discussion_r155611534) [](ancestors = 155611534)